fineract-provider/src/main/java/org/apache/fineract/portfolio/client/service/ClientReadPlatformServiceImpl.java

```diff
@@ -225,6 +228,7 @@ private String buildSqlStringFromClientCriteria(final SearchParameters searchPar
 
         String extraCriteria = "";
         if (sqlSearch != null) {
+        	SQLInjectionValidator.validateSQLInput(sqlSearch);
```

Comment:
Instead of validating input param string.... I would suggest rewriting the query as prepared statements..... Have a list to add/track a value when a sql is appended.....

For eg.... replace extraCriteria.append(" and g.external_id = ").append(ApiParameterHelper.sqlEncodeString(externalId));

with 
extraCriteria.append(" and g.external_id = ? ")
paramList.add(externalId)

convert the paramList to array and use it in the jdbcTemplate in the end while executing the query.....

I'm not sure if this approach could be used for order by clause